pygmt/clib/session.py

```diff
@@ -791,11 +792,62 @@ def put_vector(self, dataset, column, vector):
                 )
             )
 
+    def put_strings(self, dataset, column, strings):
+        """
+        Attach a numpy 1D array of dtype str as a column on a GMT dataset.
+
+        Use this function to attach string type numpy array data to a GMT
+        dataset and pass it to GMT modules. Wraps ``GMT_Put_Strings``.
+
+        The dataset must be created by :meth:`~gmt.clib.Session.create_data`
+        first. Use ``family='GMT_IS_DATASET|GMT_VIA_VECTOR'``.
+
+        .. warning::
+            The numpy array must be C contiguous in memory. If it comes from a
+            column slice of a 2d array, for example, you will have to make a
+            copy. Use :func:`numpy.ascontiguousarray` to make sure your vector
+            is contiguous (it won't copy if it already is).
+
+        Parameters
+        ----------
+        dataset : :class:`ctypes.c_void_p`
+            The ctypes void pointer to a ``GMT_Dataset``. Create it with
+            :meth:`~gmt.clib.Session.create_data`.
+        column : int
+            The column number of this vector in the dataset (starting from 0).
```

Comment:
```C
int GMT_Put_Strings (void *API, unsigned int family, void *X, char **array);
```
Unlike numeric vectors (i.e., GMT_Put_Vector), you can't put strings at a specific column. In GMT dataset, there is only a pointer to hold the string array. It has to be the "last" column. That's also why have to combine all trailing arrays into a single string array and then pass to GMT.

Reply:
Ah ok, I'll take out the column argument then.